### ISSUE TITLE
rclone rcd fails to refresh Google Drive token

### ISSUE THREAD

**Issue by vercas** [rclone/rclone#5039]
#### What is the problem you are having with rclone?

Using `rclone rcd` and, after that, three `rclone rc mount/mount` commands to mount different directories from within a crypto remote on top of Google Drive, it doesn't seem to renew the Google Drive access token:  
```
vfs reader: failed to write to cache file: open file failed: Get "https://www.googleapis.com/drive/v3/files/FooBar123?alt=media": couldn't fetch token - maybe it has expired? - refresh with "rclone config reconnect gdrive:": context canceled
```
The `rclone.conf` file doesn't have a new token in it - I still see one that expired.  

I've run the command that it tells me to, got a new token, it lasted for just one hour.  
I've also tried running with no client_{id,secret} in `rclone.conf` (as was suggested to other people on the forums) and the same happens.  

_**The problem doesn't happen when I use `rclone mount` commands (and no `rcd`).**_

#### What is your rclone version (output from `rclone version`)

```
rclone v1.54.0
- os/arch: linux/amd64
- go version: go1.15.7
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Ubuntu 20.04 for x86_64 (using stock HWE kernel, so `5.8.0-41-generic`)

####  Which cloud storage system are you using? (e.g. Google Drive)

Google Drive, with a crypto remote on top of it.

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```bash
rclone rcd \
        --config=/path/to/rclone.conf \
        --cache-dir=/path/to/rclone/cache \
        --drive-stop-on-upload-limit \
        --drive-stop-on-download-limit \
        --rc-addr=10.11.12.13:5080 \
        --rc-web-gui \
        --rc-web-gui-no-open-browser \
        --rc-web-gui-update \
        --rc-no-auth \
        --rc-user=vercas \
        --rc-pass=foobar

rclone rc --rc-addr 10.11.12.13:5080 --rc-user=vercas --rc-pass=foobar mount/mount \
        fs=gdrive-crypto:my/directory \
        mountPoint=/path/to/mountpoint/ \
        mountOpt='{
                "AllowNonEmpty": false,
                "AllowOther": true,
                "AllowRoot": false,
                "AsyncRead": true,
                "AttrTimeout": 2000000000,
                "Daemon": false,
                "DaemonTimeout": 0,
                "DebugFUSE": false,
                "DefaultPermissions": true,
                "ExtraFlags": [],
                "ExtraOptions": [],
                "MaxReadAhead": 131072,
                "NetworkMode": false,
                "NoAppleDouble": true,
                "NoAppleXattr": false,
                "VolumeName": "",
                "WritebackCache": true
        }' \
        vfsOpt='{
                "CacheMaxAge": 21600000000000,
                "CacheMaxSize": 429496729600,
                "CacheMode": 3,
                "CachePollInterval": 60000000000,
                "CaseInsensitive": false,
                "ChunkSize": 31457280,
                "ChunkSizeLimit": 209715200,
                "DirCacheTime": 600000000000,
                "DirPerms": 511,
                "FilePerms": 438,
                "GID": 1000,
                "NoChecksum": false,
                "NoModTime": false,
                "NoSeek": false,
                "PollInterval": 60000000000,
                "ReadAhead": 50000000,
                "ReadOnly": false,
                "ReadWait": 20000000,
                "UID": 1000,
                "Umask": 2,
                "WriteBack": 300000000000,
                "WriteWait": 1000000000
        }'

... and two more rclone rc mount/mount commands that are similar, and mount different directories with UID 0, GID 0, and AllowOther false this time ...
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
DEBUG : Google drive root 'crypt/foobarfoobar': Checking for changes on remote
DEBUG : gdrive1: Loaded invalid token from config file - ignoring
DEBUG : gdrive1: Token refresh failed try 1/5: context canceled
DEBUG : gdrive1: Loaded invalid token from config file - ignoring
DEBUG : gdrive1: Token refresh failed try 2/5: context canceled
DEBUG : gdrive1: Loaded invalid token from config file - ignoring
DEBUG : gdrive1: Token refresh failed try 3/5: context canceled 
DEBUG : gdrive1: Loaded invalid token from config file - ignoring
DEBUG : gdrive1: Token refresh failed try 4/5: context canceled
... lots of spam from VFS ...
DEBUG : gdrive1: Loaded invalid token from config file - ignoring
DEBUG : gdrive1: Token refresh failed try 5/5: context canceled
DEBUG : rc: "core/stats": with parameters map[]
DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletedDirs:12 deletes:0 elapsedTime:3647.074809834 errors:15 fatalError:true lastError:googleapi: Error 403: User rate limit exceeded., userRateLimitExceeded renames:0 retryError:false speed:0 transferTime:6.795015126 transfers:0]: <nil>
INFO  : Google drive root 'crypt/foobarfoobar': Change notify listener failure: Get "https://www.googleapis.com/drive/v3/changes?alt=json&fields=nextPageToken%2CnewStartPageToken%2Cchanges%28fileId%2Cfile%28name%2Cparents%2CmimeType%29%29&includeItemsFromAllDrives=true&pageSize=1000&pageToken=1036459&prettyPrint=false&supportsAllDrives=true": couldn't fetch token - maybe it has expired? - refresh with "rclone config reconnect gdrive1:": context canceled
```

**Comment by ncw** [comment]
This came up in the forum - can you try this beta? https://forum.rclone.org/t/google-drive-token-refresh-failed/22283/22?u=ncw

When the backend is created, the context from the `rc` call is being passed to it. This gets cancelled at the end of the rc request which means that the token refresher thinks its context has been cancelled too and refuses to do anything.

I've attempted to fix this here - can you give it a try?

[v1.55.0-beta.5216.e556df071.fix-backend-cancel-context](https://beta.rclone.org/branch/fix-backend-cancel-context/v1.55.0-beta.5216.e556df071.fix-backend-cancel-context/) on branch [fix-backend-cancel-context](https://github.com/rclone/rclone/tree/fix-backend-cancel-context) (uploaded in 15-30 mins)

This bit of code got re-worked in the beta so if that works, I'll backport it to the stable branch and get you to test that as well.

**Comment by vercas** [comment]
@ncw Installing this on my machine, I'll let you know in an hour or so if it's working. 😃

**Comment by vercas** [comment]
@ncw It works perfectly.

**Comment by ivandeex** [comment]
@ncw solution confirmed. can you merge to master please?

**Comment by ncw** [comment]
This is merged and released in v1.55
